Allow Running TypeScript in Strict Mode

**TypeScript Version:**
typescript@rc (typescript@2.4.0)
typescript@next (typescript@2.5.0-dev.20170614)

**Expected behavior:**
No SyntaxError.

**Actual behavior:**
```
...\path>node --version
v6.11.0

...\path>node --use_strict node_modules\typescript\bin\tsc --version
...\path\node_modules\typescript\lib\tsc.js:40882
            var arguments = node.arguments;
                ^^^^^^^^^
SyntaxError: Unexpected eval or arguments in strict mode
```

Similar to previous versions/issues:
https://github.com/Microsoft/TypeScript/issues/9449
https://github.com/Microsoft/TypeScript/issues/10755

we should also turn `--alwaysStrict` on on the compiler code base. this should have caught this:
```sh
tsc --p src\compiler\tsconfig.json --alwaysStrict

24727             const arguments = node.arguments;
                        ~~~~~~~~~

src/compiler/checker.ts(24727,19): error TS1100: Invalid use of 'arguments' in strict mode.

```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -24692,15 +24692,15 @@ namespace ts {
                 return grammarErrorOnNode(node, Diagnostics.Dynamic_import_cannot_have_type_arguments);
             }
 
-            const arguments = node.arguments;
-            if (arguments.length !== 1) {
+            const nodeArguments = node.arguments;
+            if (nodeArguments.length !== 1) {
                 return grammarErrorOnNode(node, Diagnostics.Dynamic_import_must_have_one_specifier_as_an_argument);
             }
 
             // see: parseArgumentOrArrayLiteralElement...we use this function which parse arguments of callExpression to parse specifier for dynamic import.
             // parseArgumentOrArrayLiteralElement allows spread element to be in an argument list which is not allowed as specifier in dynamic import.
-            if (isSpreadElement(arguments[0])) {
-                return grammarErrorOnNode(arguments[0], Diagnostics.Specifier_of_dynamic_import_cannot_be_spread_element);
+            if (isSpreadElement(nodeArguments[0])) {
+                return grammarErrorOnNode(nodeArguments[0], Diagnostics.Specifier_of_dynamic_import_cannot_be_spread_element);
             }
         }
     }

